src/profile-logic/call-tree.js

```diff
@@ -584,3 +586,89 @@ export function extractSamplesLikeTable(
       throw assertExhaustiveCheck(strategy);
   }
 }
+
+/**
+ * This function is extremely similar to computeCallTreeCountsAndTimings,
+ * but is specialized for converting sample counts into traced timing. Samples
+ * don't have duration information associated with them, it's mostly how long they
+ * were observed to be running. This function computes the timing the exact same
+ * way that the stack chart will display the information, so that timing information
+ * will agree. In the past, timing was computed by samplingInterval * sampleCount.
+ * This caused confusion when switching to the trace-based views when the numbers
+ * did not agree. In order to remove confusion, we can show the sample counts,
+ * plus the traced timing, which is a compromise between correctness, and consistency.
+ */
+export function computeTracedTiming(
+  samples: SamplesLikeTable,
+  { callNodeTable, stackIndexToCallNodeIndex }: CallNodeInfo,
+  interval: Milliseconds,
+  invertCallstack: boolean
+): TracedTiming | null {
+  if (samples.weightType && samples.weightType !== 'samples') {
+    // Only compute for the samples table.
+    return null;
+  }
+
+  // Compute the timing duration, which is the time between this sample and the next.
+  const weight = [];
+  for (let sampleIndex = 0; sampleIndex < samples.length - 1; sampleIndex++) {
+    weight.push(samples.time[sampleIndex + 1] - samples.time[sampleIndex]);
+  }
+  if (samples.length > 0) {
+    // Use the sampling interval for the last sample.
+    weight.push(interval);
+  }
+  const samplesWithWeight: SamplesTable = {
+    ...samples,
+    weight,
+  };
+
+  const sampleIndexToCallNodeIndex = getSampleIndexToCallNodeIndex(
+    samples.stack,
+    stackIndexToCallNodeIndex
+  );
```

Comment:
Not for this patch, more a passing-by comment: I always thought this step was unnecessary and that `_getStackSelfTimes` / `_getInvertedStackSelfTimes` could just look in `stackIndexToCallNodeIndex` themselves.